lib/templates.py

```diff
@@ -51,7 +51,9 @@ def reference_url(self):
                 return None
 
     def checkengine(self, engine=None):
-        if not engine:
+        if isinstance(engine,type(engine)):
```

Comment:
If it's an instance of itself? How could that ever be `False`? I'm pretty sure what this test is really asking is `if engine is None:`.